File: TaskManager/MyTaskManager/AddCommentCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTaskManager
{
    public class AddCommentCommand : Command
    {
        public IssueRepository repository;
        public int issueId;
        public Comment comment;

        public AddCommentCommand(IssueRepository repository, int issueId, string content, string author)
        {
            this.repository = repository;
            this.issueId = issueId;
            int commentId = repository.GetNextCommentId();
            this.comment = new Comment(commentId, content, author);
            this.Type = CommandType.Create;
            this.Description = $"Added comment to issue id {issueId}";
        }

        public override void Execute()
        {
            Issue issue = repository.GetIssue(issueId);
            if (issue != null)
            {
                issue.Comments.Add(comment);
                issue.LastModifiedDate = DateTime.Now;
                repository.UpdateIssue(issue);
            }
        }

        public override void Undo()
        {
            Issue issue = repository.GetIssue(issueId);
            if (issue != null)
            {
                foreach (var TempComment in issue.Comments)
                {
                    if (TempComment.Id == comment.Id) issue.Comments.Remove(TempComment);
                }
                
                issue.LastModifiedDate = DateTime.Now;
                repository.UpdateIssue(issue);
            }
        }
    }
}

File: Task Manager/src/AddCommentCommand.java
public class AddCommentCommand implements Command {
    private Issue issue;
    private Comment comment;
    private IssueRepository issueRepository;

    public AddCommentCommand(Issue issue, Comment comment, IssueRepository issueRepository) {
        this.issue = issue;
        this.comment = comment;
        this.issueRepository = issueRepository;
    }

    @Override
    public void execute() {
        issue.addComment(comment);
        issueRepository.updateIssue(issue);
    }

    @Override
    public void undo() {
        issue.removeComment(comment);
        issueRepository.updateIssue(issue);
    }
}